### PR TITLE
Bulk sending api

### PR DESCRIPTION
**Pull request by narekhovhannisyan**
## Motivation

Add support for Bluk Sending API to SDK.

## Changes

- Updated documentation.
- Added usage sample.
- Added unit tests.
- Implemented integration.

### CODE REVIEW

File: src/__tests__/lib/api/BulkSending.test.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        };
+
+        const result = await bulkSendingAPI.send(emailData);
+
+        expect(mock.history.post[0].url).toEqual(endpoint);
+        expect(mock.history.post[0].data).toEqual(JSON.stringify(emailData));
+        expect(result).toEqual(expectedResponseData);
+      });
+
+      it("rejects with api error.", async () => {
```
Comment (vittorius) [inline src/__tests__/lib/api/BulkSending.test.ts:71]: Seems to be a bit more descriptive

```suggestion
      it("handles an API error.", async () => {
```

File: src/__tests__/lib/api/BulkSending.test.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+          expect(mock.history.post[0].url).toEqual(endpoint);
+          expect(mock.history.post[0].data).toEqual(JSON.stringify(emailData));
+
+          if (error instanceof Error) {
+            expect(error.message).toEqual(expectedErrorMessage);
+          }
+        }
+      });
+
+      it("rejects with axios error.", async () => {
```
Comment (vittorius) [inline src/__tests__/lib/api/BulkSending.test.ts:113]: Seems to be a bit more descriptive

```suggestion
      it("handles an HTTP transport error.", async () => {
```

**Review by vittorius — COMMENTED** [review]
Aren't we going to update the changelog as well?

**Review by VladimirTaytor — APPROVED** [review]

File: src/lib/MailtrapClient.ts [hunk 3]
```diff
@@ -58,15 +62,15 @@ export default class MailtrapClient {
...
+     * Initialize APIs.
      */
     this.testingAPI = new TestingAPI(
       this.axios,
       this.testInboxId,
       this.accountId
     );
-
     this.general = new GeneralAPI(this.axios, this.accountId);
+    this.bulkSending = new BulkSendingAPI(this.axios);
```
Comment (vittorius) [inline src/lib/MailtrapClient.ts:73]: Probably let's name it `bulk`? I.e. `general`, `testing`, and `bulk`.

Comment (narekhovhannisyan) [reply src/lib/MailtrapClient.ts:73]: The name was taken from the documentation. However agree, looks much cleaner.

**Review by vittorius — APPROVED** [review]
Again, let's add a new minor version to the Changelog and describe changes

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,8 @@
+## [3.4.0] - 2024-06-10
+- Add support for Bulk product API.
+  - Refer to the [`examples`](examples) folder for code examples.
```
Comment (vittorius) [inline CHANGELOG.md:3]: Could be even

```suggestion
  - Refer to the [`examples/bulk`](examples/bulk) folder for code examples.
```

**Review by vittorius — APPROVED** [review]

**Pull request merged**